packages/rest-crud/src/crud-rest-builder.plugin.ts

```diff
@@ -0,0 +1,112 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: @loopback/rest-crud
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {bind, ControllerClass, inject} from '@loopback/core';
+import {
+  asModelApiBuilder,
+  ModelApiBuilder,
+  ModelApiConfig,
+} from '@loopback/model-api-builder';
+import {
+  ApplicationWithRepositories,
+  Class,
+  Entity,
+  EntityCrudRepository,
+} from '@loopback/repository';
+import {Model} from '@loopback/rest';
+import debugFactory from 'debug';
+import {defineCrudRepositoryClass, defineCrudRestController} from '.';
+
+const debug = debugFactory('loopback:boot:crud-rest');
+
+export interface ModelCrudRestApiConfig extends ModelApiConfig {
+  // E.g. '/products'
+  basePath: string;
+}
+
+@bind(asModelApiBuilder)
+export class CrudRestApiBuilder implements ModelApiBuilder {
+  readonly pattern: string = 'CrudRest';
+
+  build(
+    application: ApplicationWithRepositories,
+    modelClass: typeof Model & {prototype: Model},
+    cfg: ModelApiConfig,
+  ): Promise<void> {
+    const modelName = modelClass.name;
+    const config = cfg as ModelCrudRestApiConfig;
+    if (!config.basePath) {
+      throw new Error(
+        `Missing required field "basePath" in configuration for model ${modelName}.`,
+      );
+    }
+
+    if (!(modelClass.prototype instanceof Entity)) {
+      throw new Error(
+        `CrudRestController requires an Entity, Models are not supported. (Model name: ${modelName})`,
+      );
+    }
+    const entityClass = modelClass as typeof Entity & {prototype: Entity};
```

Comment:
Why is `& {prototype: Entity}` needed here? Isn't first part sufficient?

@raymondfeng @nabdelgadir   ^

Reply:
I remember seeing in [legacy-juggler-bridge](https://github.com/strongloop/loopback-next/blob/crud-rest/packages/repository/src/repositories/legacy-juggler-bridge.ts#L116), it's mentioned that "entityClass should have type "typeof T", but that's not supported by TSC" (where T extends Entity) and also it matches the type that's passed into [`defineCrudRepositoryClass`](https://github.com/strongloop/loopback-next/blob/crud-rest/packages/rest-crud/src/repository-builder.ts#L30)

Reply:
> I remember seeing in [legacy-juggler-bridge](https://github.com/strongloop/loopback-next/blob/crud-rest/packages/repository/src/repositories/legacy-juggler-bridge.ts#L116), it's mentioned that "entityClass should have type "typeof T", but that's not supported by TSC" (where T extends Entity) and also it matches the type that's passed into [`defineCrudRepositoryClass`](https://github.com/strongloop/loopback-next/blob/crud-rest/packages/rest-crud/src/repository-builder.ts#L30)

Good link. It answers my question.